src/components/GlobalPreferences/Notifications/NotificationsVerify.js

```diff
@@ -0,0 +1,211 @@
+import React, { useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
+import {
+  Box,
+  ButtonBase,
+  GU,
+  IconCheck,
+  Info,
+  LoadingRing,
+  RADIUS,
+  useTheme,
+  textStyle,
+} from '@aragon/ui'
+import { verifyEmailToken } from './notification-service-api'
+import {
+  VERIFY_SUBSECTION,
+  UnauthroizedError,
+  ExpiredTokenError,
+} from './constants'
+import notificationSvg from './notifications.svg'
+
+export function NotificationsVerify({
+  subsection,
+  onTokenChange,
+  onEmailChange,
+  navigateToNotifications,
+}) {
+  const [isFetching, setIsFetching] = useState(true)
+  const [verified, setVerified] = useState(false)
+  const [error, setError] = useState(null)
+  const theme = useTheme()
+
+  useEffect(() => {
+    // Parse token from subsection /verify/[TOKEN] -> [TOKEN]
+    const token = subsection.substring(VERIFY_SUBSECTION.length)
+    verifyEmailToken(token)
+      .then(longLivedToken => {
```

Comment:
Addressed in 0d72aef1e69fd205961849e5cce587c0ea3c3b7a

What about a `useCallback`. Can we safely assume that it will never be called if the component is unmounted? 

For example: https://github.com/aragon/aragon/blob/0d72aef1e69fd205961849e5cce587c0ea3c3b7a/src/components/GlobalPreferences/Notifications/ManageNotifications.js#L34-L64

Reply:
No, I don't think we can assume that from `useCallback()`; the problem is the promise's closure context won't change